third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -245,6 +248,40 @@ struct Transformer<'a> {
 }
 
 impl<'a> Transformer<'a> {
+    fn check_drop_for_params(&mut self, exit_state_before_execution: &ExitState) {
```

Comment:
As I said, we should not reimplement the check and error message a 2nd time. We have this code already, you duplicating a lot of code. The CopyDrop analysis should compute that these drops are needed. It currently does not because livevar info is broken for parameters.

The real bug here is that parameters should be part of the alive.before set _of the first instruction_ of a function. If they aren't used furthermore in the code, they will be automatically dropped by the code here.

So the fix is in livevar not here.

Reply:
I see your point but if a function parameter is never used in the code, it should not be in the alive.before set of the first instruction because LiveVarInfo.usages can only contain locations corresponding to the instruction, right?